scripts/cleanup-azure-resources.sh

```
@@ -258,6 +288,44 @@ find_service_principals() {
+find_resources_by_tag() {
+    local tag_filter="$1"
+    local tag_key="${tag_filter%%=*}"
+    local tag_value="${tag_filter#*=}"
+
+    print_info "Searching for Azure resources with tag '${tag_key}=${tag_value}'..." >&2
+
+    local query="Resources | where tags['${tag_key}'] == '${tag_value}' | project id, name, type, resourceGroup, subscriptionId | order by resourceGroup asc, type asc, name asc"
+
+    local resources_json
+    resources_json=$(az graph query -q "$query" -o json 2>/dev/null)
+    if [[ $? -ne 0 ]]; then
```

Shellcheck (0.11.0)

[style] Check exit code directly with e.g. 'if ! mycmd;', not indirectly with $?.

(SC2181)